inc/import/parser/class.ParserFITSingle.php

```diff
@@ -10,668 +10,134 @@
-
-		$this->gps['altitude'][]  = isset($this->Values['altitude']) ? substr($this->Values['altitude'][1], 0, -4) : 0;
-
-		$this->gps['km'][]        = isset($this->Values['distance']) ? round($this->Values['distance'][0] / 1e5, ParserAbstract::DISTANCE_PRECISION) : end($this->gps['km']);
-		$this->gps['heartrate'][] = isset($this->Values['heart_rate']) ? (int)$this->Values['heart_rate'][0] : 0;
-		$this->gps['rpm'][]       = isset($this->Values['cadence']) ? (int)$this->Values['cadence'][0] : 0;
-		$this->gps['power'][]     = isset($this->Values['power']) ? (int)$this->Values['power'][0] : 0;
-		//$this->gps['left_right'][]     = isset($this->Values['left_right_balance']) ? (int)$this->Values['left_right_balance'][0] : 0;
-
-		$this->gps['temp'][]      = isset($this->Values['temperature']) ? (int)$this->Values['temperature'][0] : 0;
-
-		$this->gps['time_in_s'][] = $time;
-
-		//Running Dynamics
-		$this->gps['groundcontact'][] = isset($this->Values['stance_time']) ? round($this->Values['stance_time'][0]/10) : 0;
-		$this->gps['oscillation'][]   = isset($this->Values['vertical_oscillation']) ? round($this->Values['vertical_oscillation'][0]/10) : 0;
-		$this->gps['groundcontact_balance'][] = isset($this->Values['stance_time_balance']) ? (int)$this->Values['stance_time_balance'][0] : 0;
-
-        // Fit developer fields
-        $this->gps['smo2_0'][] = isset($this->Values['smo2_0']) ? (int)$this->Values['smo2_0'][0] : 0;
-        $this->gps['smo2_1'][] = isset($this->Values['smo2_1']) ? (int)$this->Values['smo2_1'][0] : 0;
-        $this->gps['thb_0'][] = isset($this->Values['thb_0']) ? (int)$this->Values['thb_0'][0] : 0;
-        $this->gps['thb_1'][] = isset($this->Values['thb_1']) ? (int)$this->Values['thb_1'][0] : 0;
-
-        if ($time === $last) {
-			$this->mergeRecord();
-		}
-	}
+		if (isset($this->fitData->data_mesgs['file_creator']['software_version']))
+			$creatorDetails = 'firmware '.$this->fitData->data_mesgs['file_creator']['software_version'];
+		if (isset($this->fitData->data_mesgs['file_id']['product']))
+			$creatorDetails .= ($creatorDetails == '' ? '' : ', ') . 'product '.$this->fitData->product();
+		if (isset($this->fitData->data_mesgs['file_id']['serial_number']))
+			$creatorDetails .= ($creatorDetails == '' ? '' : ', ') . 'serial number '.$this->fitData->data_mesgs['file_id']['serial_number'];
+		$this->TrainingObject->setCreatorDetails($creatorDetails);
 
-	protected function mapDeveloperFieldsToNativeFieldsFor(array $developerFieldMapping) {
-		foreach ($developerFieldMapping as $devFieldName => $nativeData) {
-			$nativeFieldName = $nativeData[0];
-			$nativeFactor = $nativeData[1];
+		if (isset($this->fitData->data_mesgs['session']['timestamp']))
+			$this->TrainingObject->setTimestamp($this->fitData->data_mesgs['session']['timestamp']);
+		else if (isset($this->fitData->data_mesgs['file_id']['time_created']))
+			$this->TrainingObject->setTimestamp($this->fitData->data_mesgs['file_id']['time_created']);
```

Comment:
Using `$this->setTimestampAndTimezoneOffsetWithUtcFixFrom(...)` is essential. Currently, it seems that all timestamps are imported wrong. If the fitData is not able to serve the original time string to match timestamps in unit tests, we have a major issue here.